tool/speed.cc

```diff
@@ -1892,6 +1932,52 @@ static bool SpeedJitter(std::string selected) {
 }
 #endif
 
+static bool SpeedDHcheck(size_t prime_bit_length) {
+
+  TimeResults results;
+  bssl::UniquePtr<DH> dh_params(DH_new());
+  if (dh_params == nullptr) {
+    return false;
+  }
+
+  // DH_generate_parameters_ex grows exponentially slower as prime length grows.
+  if (DH_generate_parameters_ex(dh_params.get(), prime_bit_length,
```

Comment:
Does performance vary with the parameters? Should this test use a fixed dh_params per size?

Reply:
I agree it can't, but would it make the data more useful/comparable?

Reply:
It would definitely make it faster. As prime size grows, this function does exponentially more work and is not even the thing we want to measure.

Can we do it at a later point? We need some more work to actually generate static parameters.